src/reports/library/report.cpp

```diff
@@ -0,0 +1,126 @@
+#include <cmath>
+#include <iostream>
+#include <limits>
+
+#include "../utils/logger.h"
+#include "report.h"
+#include "sonatareport.h"
+
+namespace bbp {
+namespace sonata {
+
+// 4MB
+#define DEFAULT_MAX_BUFFER_SIZE 4194304
```

Comment:
does this need to be a define?  shouldn't it be a parameter? a const? constexpr?

Reply:
It can be changed via method (set_max_buffer_size()).
But it is defaulted in case the method is not called (could maybe be a default value in the constructor instead?)

Reply:
> But it is defaulted in case the method is not called (could maybe be a default value in the constructor instead?)

default value in ctor can work, but I would still make it a named constant, so that there isn't some magic number as the default.

Why did you have to increase it in `6b0a2a8`?

Reply:
Because in CoreNEURON there is a default value passed, but in Neuron, if you don't pass the buffer size in the BlueConfig it uses the default buffer defined here. Before it was 1KB (too many writings done for big compartment reports) and 4MB is a more reasonable buffer size.